### PR TITLE
Feature/regarch 49 buildings index

### PR DESCRIPTION
**Pull request by igor-kamil**
screenshot-www register-architektury local-2020-06-23-14-57-15-436

new components for:
- active-filter
- custom-select

todo:
- get buckets for current query/search - to display only available filters (faceted search)

### CODE REVIEW

File: app/Http/Controllers/BuildingController.php [hunk 1]
```diff
@@ -7,6 +7,35 @@
...
+        if (request()->filled('sort_by')) {
+            switch (request()->input('sort_by')) {
+                case 'newest':
+                    $buildings->orderBy('year_from', 'desc');
+                    break;
+                case 'oldest':
+                    $buildings->orderBy('year_from', 'asc');
+                    break;
+            }
+        }
```
Comment (eronisko) [inline app/Http/Controllers/BuildingController.php:28]:
```suggestion
        if (request()->input('sort_by') == 'newest') $buildings->orderBy('year_from', 'desc');
        if (request()->input('sort_by') == 'oldest') $buildings->orderBy('year_from', 'asc');
```

File: resources/lang/en/building.php [hunk 2]
```diff
@@ -9,4 +10,11 @@
     'project_duration_dates' => 'realization',
     'bibliography' => 'Bibliography',
     'related' => 'Related',
+
+    // index
+    'objects' => '{0} objects|[1] object|[2,*] objects',
+    'sort' => [
+        'oldest' => 'By oldest',
+        'newest' => 'By newest',
```
Comment (eronisko) [inline resources/lang/en/building.php:18]:
```suggestion
        'oldest' => 'Oldest first',
        'newest' => 'Newest first',
```

File: resources/sass/_fonts.scss [hunk 3]
```diff
@@ -0,0 +1,43 @@
+@charset "UTF-8";
+
+@font-face {
+  font-family: "regarch-font";
```
Comment (eronisko) [inline resources/sass/_fonts.scss:4]: Really cool!

I'd suggest calling it something like `regarch-icons`, but that's just minor detail.

File: app/Models/Building.php [hunk 4]
```diff
@@ -90,6 +90,9 @@ class Building extends Model
             'architect_names' => [
                 'type' => 'text',
                 'fields' => [
+                    'raw' => [
+                        'type' => 'keyword',
+                    ],
```
Comment (eronisko) [inline app/Models/Building.php:95]:
```suggestion
```

File: app/Models/Building.php [hunk 5]
```diff
@@ -108,9 +114,24 @@ class Building extends Model
             'location_gps' => [
                 'type' => 'geo_point',
             ],
+            'architects' => [
+                'type' => 'keyword',
+            ],
```
Comment (eronisko) [inline app/Models/Building.php:119]: Does this work for an array?

File: app/Models/Building.php [hunk 6]
```diff
@@ -143,6 +164,9 @@ public function getTagsAttribute()
     {
         $tags = $this->architects->pluck('full_name')->all();
         $tags[] = $this->location_city;
+        if (!empty($this->current_function)) {
+           $tags[] = $this->current_function;
+        }
         $tags[] = $this->years_span;
         return $tags;
```
Comment (eronisko) [inline app/Models/Building.php:171]: I'm not sure, but my guess is that more of these could be empty. So maybe something like this?
```suggestion
        $tags[] = $this->current_function;
        $tags[] = $this->years_span;
        return Arr::where($tags, fn ($tag) => !empty($tag));
```

File: app/Models/Building.php [hunk 7]
```diff
@@ -185,9 +209,51 @@ public function getLngLatAttribute()
...
+
+        return $array;
     }
+
+    public static function getFilterValues($payload)
+    {
+        $max_bucket_size = 200;
+        $body = (isSet($payload[0]['body'])) ? $payload[0]['body'] : [];
+
+        $body['aggs'] = [
```
Comment (eronisko) [inline app/Models/Building.php:226]: Nice, thank you :)

Just a suggestion: if you don't care about the actual search results (but just counts), you could add something like this to the query: https://github.com/SlovakNationalGallery/register-architektury/blob/7f123f4941cfeafce4176273fc7b68aced7f0441/app/Models/Architect.php#L79

File: resources/sass/_fonts.scss [hunk 8]
```diff
@@ -0,0 +1,43 @@
+@charset "UTF-8";
+
+@font-face {
+  font-family: "regarch-icons";
```
Comment (eronisko) [inline resources/sass/_fonts.scss:4]: Thanks for renaming this! 🙏

File: app/Models/Building.php [hunk 9]
```diff
@@ -108,9 +114,24 @@ class Building extends Model
...
+            ],
             'tags' => [
                 'type' => 'keyword',
             ],
+            'year_from' => [
+                'type' => 'integer',
+            ],
+            'current_function' => [
+                'type' => 'text',
+                'analyzer' => 'asciifolding_analyzer',
```
Comment (eronisko) [inline app/Models/Building.php:128]: Good thinking :)

File: app/Models/Building.php [hunk 10]
```diff
@@ -185,9 +209,51 @@ public function getLngLatAttribute()
...
             'architects',
         ]);
+        $array['architects'] = $this->architects->pluck('full_name')->all();
+
+        return $array;
     }
+
+    public static function getFilterValues($payload)
+    {
+        $max_bucket_size = 200;
```
Comment (eronisko) [inline app/Models/Building.php:223]: I'm not very impressed about how this works in elastic. If you give it a number too big, it'll probably scan through all of the docs all of the time. If you give it a number too smal, it'll not give you precise results. But I guess for our index size it doesn't matter much... 🤷 

(Just a rant, don't worry about changing this)

File: app/Http/Controllers/BuildingController.php [hunk 11]
```diff
@@ -7,6 +7,24 @@
 
 class BuildingController extends Controller
 {
+    public function index(Request $request)
```
Comment (eronisko) [inline app/Http/Controllers/BuildingController.php:10]: This action looks much better now 👍

File: app/Models/Building.php (hunk 5 above)
Comment (igor-kamil) [reply app/Models/Building.php:116]: yes :)

File: app/Models/Building.php (hunk 6 above)
Comment (igor-kamil) [reply app/Models/Building.php:168]: this is brilliant! I would just flatten the resulting array.. just in case...

**Pull request merged**